core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.java

```diff
@@ -1561,19 +1561,23 @@ private int safelyGetCurrentTab(SharedPreferences settings) {
     return Math.max(settings.getInt(TAG_CURRENT_TAB, 0), 0);
   }
 
+  /* This method restores tabs state in new launches, do not modify it
+     unless it is explicitly mentioned in the issue you're fixing */
   protected void restoreTabs(@Nullable String zimArticles, @Nullable String zimPositions,
     int currentTab) {
     try {
       JSONArray urls = new JSONArray(zimArticles);
       JSONArray positions = new JSONArray(zimPositions);
-      webViewList.clear();
-      currentWebViewIndex=0;
-      tabsAdapter.notifyItemRemoved(0);
-      tabsAdapter.notifyDataSetChanged();
```

Comment:
Are you sure we don't want to notify data set changed? Won't this create some regression?

Reply:
Well, it didn't create any ill effects when I ran it. I'll stress test it and get back to you

Reply:
100% sure no regressions :+1: You may check on your own as well

Reply:
This won't be a regression and the application without this as well runs fine. But I believe it would be a good practice to notify the tabsAdapter of the changes taking place.

Reply:
What about `notifyItemRemoved`, can you do `git blame` on this file and check why was it added.

Reply:
I need not blame it, I remember it very well, it was done as we got the user feedback via stephane that on creating newtabs, the previously existing ones are sent back to their homePage; I'll explain it here:
```kotlin
      currentWebViewIndex=0;
      tabsAdapter.notifyItemRemoved(0);
      tabsAdapter.notifyDataSetChanged();
```


These three lines do not let the user to come back on home page when he creates a new tab. And these calls on adapter saves the state of previously opened tab which wasn't the case when these calls on tabsAdapter were not present, at that time, on clicking the plus button(with multiple open) created a new tab and opened the home page of the book on each and every tab irrespective of their history/previous state. 
Sean handled this issue.